### PR TITLE
Added some points about MVC, Data Binding and Naming Conventions

### PR DESCRIPTION
**Pull request by waldpark**
Hi all,

I would like to contribute. 

Thanks and best regards,
Adrian

### CODE REVIEW

File: docs/mvc/baseController.markdown [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# Use base controllers
+
+Keep commonly used methods such as
+
+- getResourceBundle
+- getRouter
+- getModel
+- setModel
+- onNavBack
+- _or whatever method or property is being actively re-used inside your project_
```
Comment (vobu) [inline docs/mvc/baseController.markdown:18]: wording suggestion: "method or property is being re-used in other controllers in the application".
but YMMV :)

File: docs/index.markdown [hunk 2]
```diff
@@ -9,7 +9,7 @@ layout: home
 
 - [Error Handling (Tobias)](#blank)
 - [i18n (Marian)](i18n)
-- [Naming Convetion (tbd)](#blank)
+- [Naming Convention (Adrian)](nc)
```
Comment (vobu) [inline docs/index.markdown:12]: cool! 🥳

File: docs/index.markdown [hunk 3]
```diff
@@ -18,9 +18,12 @@ layout: home
 - [oData v2,v4, REST (tbd)](#blank)
   - Do and Don'ts
   - Don't: Plain Ajax Requests
+- [Model-View-Controller (Adrian)](mvc)
```
Comment (vobu) [inline docs/index.markdown:21]: cool! 🥳

File: docs/mvc/dataBinding.markdown [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+layout: default
+title: Data Binding
+name: Overview
+parent: Model View Controller
+nav_order: 2
+---
+
+# Make use Data Binding
```
Comment (vobu) [inline docs/mvc/dataBinding.markdown:9]: for this entire chapter alone I'd like to 🤗 you in the name of the entire UI5 dark matter developers;
still: "make use _of_ data binding"

**Review by vobu — CHANGES_REQUESTED** [review]
there is so much good stuff in here, THANKS!
only some minor suggestions, otherwise a super-welcome contribution.
keep it up!!!

File: docs/mvc/baseController.markdown (hunk 1 above)
Comment (waldpark) [reply docs/mvc/baseController.markdown]: done

File: docs/mvc/dataBinding.markdown (hunk 4 above)
Comment (waldpark) [reply docs/mvc/dataBinding.markdown]: done

**Comment by waldpark** [comment]
Please check my fixes. Thanks!

File: docs/mvc/baseController.markdown [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+inside a base controller:
+
+    |-+ webapp
+      |-+ controller
+        |-- BaseController.js
+
+**webapp/controller/BaseController.js:**
+
+    sap.ui.define([
```
Comment (vobu) [inline docs/mvc/baseController.markdown:28]: sorry, missed this in the last review: please add a codefence à la ```js to get syntax highlighting:
```js
const hello = () => { world: "yay" }
```

File: docs/mvc/baseController.markdown [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+      });
+    });
+
+All view controllers should use the _BaseController.js_ as their parent controller like so:
+
+**webapp/controller/App.controller.js:**
+
+    sap.ui.define([
```
Comment (vobu) [inline docs/mvc/baseController.markdown:59]: same here: please add a codefence à la ```js to get syntax highlighting

File: docs/mvc/dataBinding.markdown [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+name: Overview
+parent: Model View Controller
+nav_order: 2
+---
+
+# Make use of Data Binding
+
+Instead of manipulating the behavior of your control
+
+    <Title text="Hello {/recipient/firstName} {/recipient/lastName}" id="myControl123" />
```
Comment (vobu) [inline docs/mvc/dataBinding.markdown:13]: please add ```xml code fence

File: docs/mvc/dataBinding.markdown [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+by its id, like so,
+
+    var oTitle = this.getView().byId("myControl123");
```
Comment (vobu) [inline docs/mvc/dataBinding.markdown:17]: same here: please add a codefence à la ```js to get syntax highlighting

File: docs/mvc/dataBinding.markdown [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    oTitle.setText("Hello John Doe");
+
+better make use of data binding
+
+    <Title text="Hello {DisplayModel>/recipient/firstName} {DisplayModel>/recipient/lastName}"/>
```
Comment (vobu) [inline docs/mvc/dataBinding.markdown:22]: please add ```xml code fence

File: docs/mvc/dataBinding.markdown [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+and work on the data model:
+
+    var oDisplayModel = this.getView().getModel("DisplayModel");
```
Comment (vobu) [inline docs/mvc/dataBinding.markdown:26]: same here: please add a codefence à la ```js to get syntax highlighting

File: docs/nc/namespaces.markdown [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+layout: default
+title: Namespaces
+name: Overview
+parent: Naming Conventions
+nav_order: 1
+---
+
+# XML Views
+
+    <mvc:View
```
Comment (vobu) [inline docs/nc/namespaces.markdown:11]: please add XML code fence: ```xml

File: docs/nc/modelNaming.markdown [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+name: Overview
+parent: Naming Conventions
+nav_order: 2
+---
+
+# Model Naming
+
+The standard view model is very often unnamed in all the Walkthrough tutorials:
+
+    var oData = {
```
Comment (vobu) [inline docs/nc/modelNaming.markdown:13]: same here: please add a codefence à la ```js to get syntax highlighting

File: docs/nc/modelNaming.markdown [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        firstName : "John",
+        lastName : "Doe"
+      }
+    };
+    var oModel = new JSONModel(oData);
+    this.getView().setModel(oModel); // <-- No naming here!
+
+Then, inside the corresponding view XML, model data may be referenced like so:
+
+    <Title text="Hello {/recipient/firstName} {/recipient/lastName}" />
```
Comment (vobu) [inline docs/nc/modelNaming.markdown:24]: please add ```xml code fence

File: docs/nc/modelNaming.markdown [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+In large projects with diverse models inside many views and controllers, its often very helpful to quickly lookup the usage of a certain model. In such cases it would really save a lot time if the model had a proper naming:
+
+    this.getView().setModel(oModel, "DisplayModel"); // <-- Example of a proper name
```
Comment (vobu) [inline docs/nc/modelNaming.markdown:28]: same here: please add a codefence à la ```js to get syntax highlighting

File: docs/nc/modelNaming.markdown [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+Then, inside the view XML, it would be:
+
+    <Title text="Hello {DisplayModel>/recipient/firstName} {DisplayModel>/recipient/lastName}" />
```
Comment (vobu) [inline docs/nc/modelNaming.markdown:32]: please add ```xml code fence

**Review by vobu — CHANGES_REQUESTED** [review]
sorry, i missed hinting on using code fences for JS- + XML-syntax highlighting in the last review.
tried pointing out all places where to add them - in case you come across some i missed, please add them also

File: docs/mvc/baseController.markdown (hunk 5 above)
Comment (waldpark) [reply docs/mvc/baseController.markdown]: Done for all code snippets. Please check again!

**Review by vobu — APPROVED** [review]
great stuff 🙌

**Pull request merged**